src/main/rx/fport.c

```diff
@@ -25,7 +25,7 @@
 
 #include "platform.h"
 
-#if defined(USE_SERIAL_RX)
+#if defined(USE_SERIAL_RX) && defined(USE_SERIALRX_FPORT)
```

Comment:
I took a look at this and put in:
```
#if !defined(USE_SERIAL_RX)
#undef USE_SERIALRX_CRSF
#undef USE_SERIALRX_IBUS
#undef USE_SERIALRX_SBUS
#undef USE_SERIALRX_SPEKTRUM
#undef USE_SERIALRX_SUMD
#undef USE_SERIALRX_SUMH
#undef USE_SERIALRX_XBUS
#undef USE_SERIALRX_FPORT
#endif
```
And unsurprisingly that breaks a lot because many of the telemetry protocols are based on serial rx protocols.  So the question is whether there are any telemetry types we want to leave enabled if `USE_SERIAL_RX` is not defined or do all depend on serial rx?

Also I missed a couple of telemetry types in the first commit (surprisingly they compile with `#undef USE_TELEMETRY`:
```
#define USE_TELEMETRY_IBUS
#define USE_TELEMETRY_IBUS_EXTENDED
#define USE_TELEMETRY_JETIEXBUS
#define USE_TELEMETRY_MAVLINK
```
So the final section for `#if !defined(USE_TELEMETRY)` would be:
```
#if !defined(USE_TELEMETRY)
#undef USE_CRSF_CMS_TELEMETRY
#undef USE_TELEMETRY_CRSF
#undef USE_TELEMETRY_FRSKY_HUB
#undef USE_TELEMETRY_HOTT
#undef USE_TELEMETRY_IBUS
#undef USE_TELEMETRY_IBUS_EXTENDED
#undef USE_TELEMETRY_JETIEXBUS
#undef USE_TELEMETRY_LTM
#undef USE_TELEMETRY_MAVLINK
#undef USE_TELEMETRY_SMARTPORT
#undef USE_TELEMETRY_SRXL
#undef USE_SERIALRX_FPORT
#endif
```
So the question is if `!defined(USE_SERIAL_RX)` should we also always disable `USE_TELEMETRY`?  Or are there some that are still valid without serial rx?  Maybe `USE_TELEMETRY_FRSKY_HUB` and `USE_TELEMETRY_MAVLINK`?

Reply:
Ok, I'll just fix up the specific telemetry dependencies to avoid compile errors or warnings if `!defined(USE_SERIAL_RX)`.